[test visibility] Add known limitation to auto test retries for Javascript (#24598)
DataDog/documentation: content/en/tests/auto_test_retries/_index.md

The default behavior of the feature is to retry any failing test case up to 5 times.

#### Known limitations

> [jest-image-snapshot][2] is incompatible with `jest.retryTimes` unless `customSnapshotIdentifier` is passed (see [jest-image-snapshot docs][3]) to `toMatchImageSnapshot`, so auto test retries will not work unless `customSnapshotIdentifier` is used.

Reviewer: ```suggestion
[jest-image-snapshot][2] is incompatible with `jest.retryTimes` unless `customSnapshotIdentifier` is passed (see [jest-image-snapshot docs][3]) to `toMatchImageSnapshot`. Therefore, auto test retries do not work unless `customSnapshotIdentifier` is used.
```
